test/test_resume.cpp

```diff
@@ -1690,3 +1690,42 @@ TORRENT_TEST(resume_data_have_pieces)
 	TEST_EQUAL(rs->params.unfinished_pieces.size(), 1);
 }
 
+// See https://github.com/arvidn/libtorrent/issues/5174
+TORRENT_TEST(removed)
+{
+	// we're _likely_ to trigger the condition, but not guaranteed. loop
+	// until we do.
+	while (true) {
+		lt::session ses(settings());
+		std::shared_ptr<torrent_info> ti = generate_torrent();
+		add_torrent_params p;
+		p.ti = ti;
+		p.save_path = ".";
+		torrent_handle h = ses.add_torrent(p);
+		// this is asynchronous
+		ses.remove_torrent(h);
+		try {
+			h.save_resume_data();
+		} catch (std::exception const& e) {
+			std::printf("failed to trigger condition, retrying\n");
+			continue;
+		}
+		// we triggered the condition.
+		while (true)
+		{
+			std::vector<alert*> alerts;
+			alert const* ap = ses.wait_for_alert(seconds(1));
+			TEST_CHECK(ap != nullptr);
+			if (ap == nullptr) return;
```

Comment:
I suppose it's not infinite, since it exists here. I think it would be better to type this out as a loop with this as the termination condition, to make it easier to read.